### PR TITLE
Add config variable and check for platform warnings

### LINKED ISSUE #6124: The ability to silence platform warnings
https://github.com/bundler/bundler/blob/b9c12ee67661581f8a4fb5f47a375b5c7da2ed1e/lib/bundler/definition.rb#L880-L884

There really isn't any reason to ship that warning to me, and it would be nice if I could silence it, as I am well aware of the platform I am on, and I always inspect Gemfiles before I install anything, regardless of whether it's my own project or not.

### PR DESCRIPTION
**Pull request by agrim123**
Thanks so much for the contribution!
To make reviewing this PR a bit easier, please fill out answers to the following questions.

### What was the end-user problem that led to this PR?

The user needed a way to turn off platform warnings.

### What was your diagnosis of the problem?

Creating a config variable to solve the above problem.

### What is your fix for the problem, implemented in this PR?

Added a key `disable_platform_warnings` in settings and placed check at the relevant place to disable warnings.

### Why did you choose this fix out of the possible options?

We will by default show warnings but the user might want to disable them, so using a config variable looked a good option.

Fixes #6124

### CODE REVIEW

**Comment by colby-swandale** [comment]
Thanks for helping make Bundler better! A couple of things that need to be done before we can merge this. This is going to need a test to make sure this is behaving as expected and so that we don't regress in the future, you will want to look at [this test file](https://github.com/bundler/bundler/blob/master/spec/install/gemfile/platform_spec.rb#L233). You will also need to document the new configuration option to the [bundle config man page](https://github.com/bundler/bundler/blob/master/man/bundle-config.ronn#L116)

File: spec/install/gemfile/platform_spec.rb [hunk 1]
```diff
@@ -230,21 +230,46 @@
     expect(out).not_to match(/Could not find gem 'some_gem/)
   end
 
-  it "prints a helpful warning when a dependency is unused on any platform" do
-    simulate_platform "ruby"
-    simulate_ruby_engine "ruby"
+  context "when disable_platform_warnings is false (by default)" do
```
Comment (colby-swandale) [inline spec/install/gemfile/platform_spec.rb]: This test doesn't need a `context` blocked wrapping around it as the setting is set in Bundler automatically.

File: spec/install/gemfile/platform_spec.rb [hunk 2]
```diff
@@ -230,21 +230,46 @@
...
+      expect(out).to include <<-O.strip
+  The dependency #{Gem::Dependency.new("rack", ">= 0")} will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
+      O
+    end
+  end
+
+  context "when disable_platform_warnings is true" do
+    before { bundle! "config disable_platform_warnings true" }
+
+    it "doesnot print the warning when a dependency is unused on any platform" do
```
Comment (colby-swandale) [inline spec/install/gemfile/platform_spec.rb]: `does not`

File: spec/install/gemfile/platform_spec.rb [hunk 3]
```diff
@@ -230,21 +230,46 @@
...
+      gemfile <<-G
+        source "file://#{gem_repo1}"
+
+        gem "rack", :platform => [:mingw, :mswin, :x64_mingw, :jruby]
+      G
+
+      bundle! "install"
+
+      expect(out).not_to include <<-O.strip
+  The dependency #{Gem::Dependency.new("rack", ">= 0")} will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
```
Comment (colby-swandale) [inline spec/install/gemfile/platform_spec.rb]: This is a lot to match exactly against in the output which could easily be missed if we change the test or behavior. I would match with:

    expect(out).to_not include(/The dependency (.*) will be unused/)

File: spec/install/gemfile/platform_spec.rb (hunk 2 above)
Comment (agrim123) [reply spec/install/gemfile/platform_spec.rb]: My bad!

File: spec/install/gemfile/platform_spec.rb (hunk 3 above)
Comment (agrim123) [reply spec/install/gemfile/platform_spec.rb]: Cool. Will change it.

Comment (colby-swandale) [reply spec/install/gemfile/platform_spec.rb]: Yep, that was a typo on my part.

Comment (agrim123) [reply spec/install/gemfile/platform_spec.rb]: Using 
```bash
expect(out).not_to match(/The dependency (.*) will be unused/)
```
does the trick. Is it good?

File: man/bundle-config.ronn [hunk 4]
```diff
@@ -166,6 +166,8 @@ learn more about their operation in [bundle install(1)][bundle-install(1)].
    When set, Gemfiles containing multiple sources will produce errors
    instead of warnings.
    Use `bundle config --delete disable_multisource` to unset.
+* `disable_platform_warnings` (`BUNDLE_DISABLE_PLATFORM_WARNINGS`):
+   Whether Bundler should show platform warnings.
```
Comment (colby-swandale) [inline man/bundle-config.ronn]: I would reword this to be a bit more explicit, something like:

```
Disable warnings during bundle install when a dependency is unused on the current platform
```

Comment (agrim123) [reply man/bundle-config.ronn]: Done :smile:

**Comment by colby-swandale** [comment]
Thanks! @bundlerbot r+
